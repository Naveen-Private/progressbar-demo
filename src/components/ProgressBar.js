import React from 'react'
import styled from 'styled-components'

import { media } from '../assets/appCSS'

const ProgressBarDiv = styled.div`
    display: flex;
    width: 650px;
    height: 50px;
    border: 1px solid #ccc;
    border-radius: 2px;
    margin: 18px;
    ${media.mobile`
        width: 100%;
    `}
`

const Progress = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: ${props => props.width || 0}%;
    height: inherit;
    background-color: ${props => props.width > 100 ? 'red' : 'green'};
`

const Value = styled.div`
    position: absolute;
    left: 49%;
    align-self: center;
`

class ProgressBar extends React.Component {
    render (){
        const {
            value
        } = this.props
        return (
            <ProgressBarDiv>
                <Value>{value > 0 ? value : '0'}%</Value>
                <Progress width={value > 0 ? value : 0} />
            </ProgressBarDiv>
        )
    }
}

export default ProgressBar