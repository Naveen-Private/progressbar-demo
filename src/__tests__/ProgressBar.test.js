import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ProgressBar from '../components/ProgressBar'

describe('ProgressBar', () => {
    it('should be defined', () => {
      expect(ProgressBar).toBeDefined();
    });
    it('should render correctly', () => {
      const tree = shallow(
        <ProgressBar value='100' />
      );
      expect(tree).toMatchSnapshot();
    });
});

