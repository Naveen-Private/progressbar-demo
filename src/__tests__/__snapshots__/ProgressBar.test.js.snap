// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    value="100"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          100
          %
</styled.div>,
        <styled.div
          width="100"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "100",
            "%",
          ],
        },
        "ref": null,
        "rendered": Array [
          "100",
          "%",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "width": "100",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            100
            %
</styled.div>,
          <styled.div
            width="100"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "100",
              "%",
            ],
          },
          "ref": null,
          "rendered": Array [
            "100",
            "%",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "width": "100",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
